src/network/address.rs

```diff
@@ -110,12 +110,186 @@ impl fmt::Debug for Address {
     }
 }
 
+/// Address received from BIP155 addrv2 message
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct AddrV2Address {
+    /// Time that this node was last seen as connected to the network
+    pub time: u32,
+    /// Service bits
+    pub services: ServiceFlags,
+    /// IP address
```

Comment:
Not necessary an IP address, I think it should say "Network ID + Network address. The interpretation depends on networkID". Just copy the BIP.